### PR TITLE
Rename Polygon -> PolySurface

### PR DESCRIPTION
**Pull request by serenity4**
Everything should be renamed properly from `Polygon` to `PolySurface`, as suggested in #6.
I also updated the docstrings and variable names to "polysurface" instead of "polygon" when it referenced the new `PolySurface` type.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 *.jl.mem
 .DS_Store
 Manifest.toml
+/.vscode
```
Comment (juliohm) [inline .gitignore]: I think you can remove the `/` as it is assumed by default

File: src/Meshes.jl [hunk 2]
```diff
@@ -74,7 +74,7 @@ export
   windingnumber, orientation,
 
   # polygons
```
Comment (juliohm) [inline src/Meshes.jl]: Missed comment update

File: src/discretization/fist.jl [hunk 3]
```diff
@@ -23,7 +23,7 @@ to heuristics implemented in the algorithm.
 """
 struct FIST <: DiscretizationMethod end
 
-function discretize(polygon::Polygon, method::FIST)
+function discretize(polygon::PolySurface, method::FIST)
```
Comment (juliohm) [inline src/discretization/fist.jl]: We should also update the argument to `polysurface`

File: src/polygons.jl [hunk 4]
```diff
@@ -3,9 +3,9 @@
 # ------------------------------------------------------------------
 
 """
-    Polygon(outer, [inner1, inner2, ..., innerk])
+    PolySurface(outer, [inner1, inner2, ..., innerk])
 
-A polygon with `outer` ring, and optional inner
+A polygon surface with `outer` ring, and optional inner
```
Comment (juliohm) [inline src/polygons.jl]: I would write "A polygonal surface"

**Review by juliohm — CHANGES_REQUESTED** [review]
Awesome. Thanks for the quick PR! Just a few small changes before we can merge it.

**Comment by serenity4** [comment]
Ah, I missed those. Thanks!

File: src/plotrecipes/polygons.jl [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
-@recipe function f(polygon::Polygon)
+@recipe function f(polygon::PolySurface)
```
Comment (juliohm) [inline src/plotrecipes/polygons.jl]: Missed this one. I would do a /search/replace in the codebase to update every instance, including the tests.

**Comment by serenity4** [comment]
Right, it should be fine now.

**Pull request merged**